src/java/org/jivesoftware/openfire/muc/spi/MUCPersistenceManager.java

```diff
@@ -129,8 +129,8 @@
     private static final String DELETE_USER_MUCAFFILIATION =
         "DELETE FROM ofMucAffiliation WHERE jid=?";
     private static final String ADD_CONVERSATION_LOG =
-        "INSERT INTO ofMucConversationLog (roomID,sender,nickname,logTime,subject,body,stanza) " +
-        "VALUES (?,?,?,?,?,?,?)";
+        "INSERT INTO ofMucConversationLog (roomID,messageID,sender,nickname,logTime,subject,body,stanza) " +
+        "SELECT ?,COUNT(*),?,?,?,?,?,? FROM ofMucConversationLog";

```

Comment:
Yeah, but I worry the count(*) approach will be very resource expensive for the DB to run.  Not all DBs are optimized for this query to be fast and requires a full table sequence scan.  I'll investigate this some more now.